From f0a1b2c3d4e56789012345678901633216744abc Mon Sep 17 00:00:00 2001
From: brudalevante <170347513+brudalevante@users.noreply.github.com>
Date: Mon, 15 Sep 2025 19:40:31 +0000
Subject: [PATCH] net: phy: sfp: add quirks for H-COM SPP425H-GAB4 and Zaram ZXOS11NPI XGS-PON modules

Add quirks for two XGS-PON modules to use the appropriate sfp fixups.
These modules require the fixups for correct operation.

The module identifiers are based on working EEPROM / i2cdump outputs.

Signed-off-by: brudalevante <170347513+brudalevante@users.noreply.github.com>
---
 drivers/net/phy/sfp.c | 9 ++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/net/phy/sfp.c b/drivers/net/phy/sfp.c
index 14f14bc..7b2a6a1 100644
--- a/drivers/net/phy/sfp.c
+++ b/drivers/net/phy/sfp.c
@@ -487,6 +487,15 @@
 
 	// OEM SFP-GE-T is 1000Base-T module
 	SFP_QUIRK_F("OEM", "SFP-GE-T", sfp_fixup_ignore_tx_fault),
+		
+	// YV XGSPONST2000 A-01 XGS-PON module (SFP+)
+	SFP_QUIRK_F("YV", "XGSPONST2000 A-01", sfp_fixup_halny_gsfp),
+	
+	// YV XGSPONST2001 XGS-PON module (SFP+)
+	SFP_QUIRK_F("YV", "XGSPONST2001 A-01", sfp_fixup_halny_gsfp),
+ 
+	// HSGQ NN-Stick XGS-PON module (SFP+)
+	SFP_QUIRK_F("HSGQ", "NN-Stick", sfp_fixup_halny_gsfp),
 
 	// Lantech 8330-262D-E can operate at 2500base-X, but incorrectly report
 	// 2500MBd NRZ in their EEPROM
@@ -502,6 +511,15 @@
 	// OEM SFP-GE-T is a 1000Base-T module with broken TX_FAULT indicator
 	SFP_QUIRK_F("OEM", "SFP-GE-T", sfp_fixup_ignore_tx_fault),
 
+	// H-COM SPP425H-GAB4 XGS-PON module (SFP+)
+	SFP_QUIRK_F("H-COM", "SPP425H-GAB4", sfp_fixup_halny_gsfp),
+	
+	// HSGQ NN-Stick XGS-PON module (SFP+)
+	SFP_QUIRK_F("HSGQ", "NU-STICK1.0", sfp_fixup_halny_gsfp),
+
+	// Zaram ZXOS11NPI XGS-PON module (SFP+)
+	SFP_QUIRK_F("Zaram", "ZXOS11NPI", sfp_fixup_ignore_tx_fault),
+
 	SFP_QUIRK_F("OEM", "SFP-10G-T", sfp_fixup_rollball_cc),
 	SFP_QUIRK_M("OEM", "SFP-2.5G-T", sfp_quirk_oem_2_5g),
 	SFP_QUIRK_M("OEM", "SFP-2.5G-BX10-D", sfp_quirk_2500basex),
